test/spec/modules/greenbidsBidAdapter_specs.js

```diff
@@ -0,0 +1,964 @@
+      let bidderRequest = {
+        'auctionId': '1d1a030790a475',
+        'bidderRequestId': '22edbae2733bf6',
+        'timeout': 3000,
+        'gppConsent': {
+          'gppString': undefined,
+          'applicableSections': ['a']
+        }
+      };
+
+      const request = spec.buildRequests(bidRequests, bidderRequest);
+      const payload = JSON.parse(request.data);
+
+      expect(payload.gpp).to.exist;
+      expect(payload.gpp.consentString).to.equal('');
+      expect(payload.gpp.applicableSectionIds).to.have.members([]);
+    });
+
+    it('should not set the GPP object in the request sent to the endpoint when not present', function () {
+      let bidderRequest = {
+        'auctionId': '1d1a030790a475',
+        'bidderRequestId': '22edbae2733bf6',
+        'timeout': 3000
+      };
+
+      const request = spec.buildRequests(bidRequests, bidderRequest);
+      const payload = JSON.parse(request.data);
+
+      expect(payload.gpp).to.not.exist;
+    });
+
+    it('should send GDPR to endpoint', function () {
+      let consentString = 'JRJ8RKfDeBNsERRDCSAAZ+A==';
+      let bidderRequest = {
+        'auctionId': '1d1a030790a475',
+        'bidderRequestId': '22edbae2733bf6',
+        'timeout': 3000,
+        'gdprConsent': {
+          'consentString': consentString,
+          'gdprApplies': true,
+          'vendorData': {
+            'isServiceSpecific': true
+          },
+          'apiVersion': 2
+        }
+      };
+
+      const request = spec.buildRequests(bidRequests, bidderRequest);
+      const payload = JSON.parse(request.data);
+
+      expect(payload.gdpr_iab).to.exist;
+      expect(payload.gdpr_iab.consent).to.equal(consentString);
+      expect(payload.gdpr_iab.status).to.equal(12);
+      expect(payload.gdpr_iab.apiVersion).to.equal(2);
+    });
+
+    it('should add referer info to payload', function () {
+      const bidRequest = Object.assign({}, bidRequests[0])
+      const bidderRequest = {
+        refererInfo: {
+          page: 'https://example.com/page.html',
+          reachedTop: true,
+          numIframes: 2
+        }
+      }
+      const request = spec.buildRequests([bidRequest], bidderRequest);
+      const payload = JSON.parse(request.data);
+
+      expect(payload.referrer).to.exist;
+      expect(payload.referrer).to.deep.equal('https://example.com/page.html')
+    });
+
+    it('should add networkBandwidth info to payload', function () {
+      const request = spec.buildRequests(bidRequests, bidderRequestDefault);
+      const payload = JSON.parse(request.data);
+
+      const bandwidth = window.navigator && window.navigator.connection && window.navigator.connection.downlink;
+
+      expect(payload.networkBandwidth).to.exist;
+
+      if (bandwidth) {
+        expect(payload.networkBandwidth).to.deep.equal(bandwidth.toString());
+      } else {
+        expect(payload.networkBandwidth).to.deep.equal('');
+      }
```

Comment:
thought: this is a weird test where you don't know what to check ? I suppose that there is no way to mock the `window.navigator...` to check the right access ?